### PR TITLE
delete operation is added

### PR DESCRIPTION
**Pull request by anil-intelops**

### CODE REVIEW

File: capten/agent/pkg/capten-store/tekton_projects.go [hunk 1]
```diff
@@ -72,7 +72,23 @@ func (a *Store) updateTektonProjects() ([]*model.TektonProject, error) {
...
 	for _, tekPro := range regTektonProjects {
-		regTektonProjectId[tekPro.Id] = tekPro
+		var deleteRecord bool
+		for _, gitProject := range allTektonProjects {
+			if gitProject.Id == tekPro.GitProjectId {
+				deleteRecord = false
+				break
+			} else {
+				deleteRecord = true
+			}
```
Comment (vramk23) [inline capten/agent/pkg/capten-store/tekton_projects.go:82]: else block not needed

File: capten/agent/pkg/capten-store/crossplane_project.go [hunk 2]
```diff
@@ -74,6 +74,24 @@ func (a *Store) updateCrossplaneProject() (*model.CrossplaneProject, error) {
...
+	for _, crossplaneProject := range regCrossplaneProjects {
+		var deleteRecord bool
+		for _, gitProject := range allCrossplaneGitProjects {
+			if gitProject.Id == crossplaneProject.GitProjectId {
+				deleteRecord = false
+				break
+			} else {
+				deleteRecord = true
+			}
+		}
```
Comment (vramk23) [inline capten/agent/pkg/capten-store/crossplane_project.go:86]: else block not needed

File: capten/agent/pkg/capten-store/argocd_projects.go [hunk 3]
```diff
@@ -102,8 +102,24 @@ func (a *Store) updateArgoCDProjects() ([]*model.ArgoCDProject, error) {
...
-		argoCDProjects[tekPro.Id] = tekPro
+	for _, argoCDPro := range regArgoCDProjects {
+		var deleteArgoCDRecord bool
+		for _, gitProject := range gitProjects {
+			if gitProject.Id == argoCDPro.GitProjectId {
+				deleteArgoCDRecord = false
+				break
+			} else {
+				deleteArgoCDRecord = true
+			}
```
Comment (vramk23) [inline capten/agent/pkg/capten-store/argocd_projects.go:113]: else block not needed

File: capten/agent/pkg/capten-store/argocd_projects.go [hunk 4]
```diff
@@ -102,8 +102,24 @@ func (a *Store) updateArgoCDProjects() ([]*model.ArgoCDProject, error) {
...
+		if deleteArgoCDRecord {
+			if err := a.DeleteArgoCDProjectsData(argoCDPro.Id); err != nil {
+				return nil, err
+			}
+		} else {
+			argoCDProjects[argoCDPro.GitProjectId] = argoCDPro
+		}
 	}
 
 	ret := make([]*model.ArgoCDProject, 0)
```
Comment (vramk23) [inline capten/agent/pkg/capten-store/argocd_projects.go:125]: check first if exist in map, if exist use ap project from the map,
if not exist, create and add the project.

do the same for other methosds

project := &model.ArgoCDProject{Id: gitProject.Id, GitProjectId: gitProject.Id,
			GitProjectUrl: gitProject.ProjectUrl}
		if ap, ok := argoCDProjects[gitProject.Id]; !ok {
			project.Status = string(model.ArgoCDProjectAvailable)
			project.LastUpdateTime = time.Now().Format(time.RFC3339)
			if err := a.UpsertArgoCDProject(project); err != nil {
				return nil, err
			}
		} else {
			project.Status = argoCDProjects[gitProject.Id].Status
			project.LastUpdateTime = ap.LastUpdateTime
		}

File: capten/agent/pkg/agent/plugin_git_apis.go [hunk 5]
```diff
@@ -132,6 +132,13 @@ func (a *Agent) DeleteGitProject(ctx context.Context, request *captenpluginspb.D
 	}
 	a.log.Infof("Delete Git project %s request recieved", request.Id)
```
Comment (vramk23) [inline capten/agent/pkg/agent/plugin_git_apis.go:133]: missing vault cred deletion for cloud provider delete api

**Review by vramk23 — CHANGES_REQUESTED** [review]

File: capten/agent/pkg/capten-store/tekton_projects.go (hunk 1 above)
Comment (anil-intelops) [reply capten/agent/pkg/capten-store/tekton_projects.go:82]: done

**Pull request merged**